src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -191,6 +222,23 @@ internal static ScriptHostConfiguration CreateScriptHostConfiguration(WebHostSet
             return scriptHostConfig;
         }
 
+        private TraceWriter CreateDefaultTraceWriter(WebHostSettings settings)
+        {
+            // need to set up a default trace writer that logs both host logs and system logs
+            var config = GetScriptHostConfiguration(settings);
+            var systemEventGenerator = config.HostConfig.GetService<IEventGenerator>() ?? new EventGenerator();
+            TraceWriter systemTraceWriter = new SystemTraceWriter(systemEventGenerator, _settingsManager, TraceLevel.Verbose);
+
+            // Note that we're creating a logger here that is independent of log configuration settings
+            // since we haven't read config yet. This logger is independent on the host having been started.
+            // That does mean that even if file logging is disabled, some logs might get written to the file system
+            // but that's ok.
+            string hostLogFilePath = Path.Combine(config.RootLogPath, "Host");
+            TraceWriter fileTraceWriter = new FileTraceWriter(hostLogFilePath, config.HostConfig.Tracing.ConsoleLevel);
```

Comment:
Once the Host starts, this means that we'll have created two Host FileTraceWriters. Internally, they take locks around the write operations -- could that be an issue here since now we'll have two instances and those locks won't be synchronized?

I don't think we'll ever be using both of these writers at the same time, but I seem to remember bumping into this when I was originally plumbing the ILogger through.

Reply:
Before these changes, it's already the case that multiple FileTraceWriters are in play and active at the same time. E.g. when hosts are spinning down and being created concurrently, the StructuredLogWriter also creates a FileTraceWriter, etc. I don't think this is an issue (or at least not a new issue)

Reply:
Yeah, makes sense. I think the issue I was having was the same host trying to flush to two FileTraceWriters pointing to the same file -- at the same time. Just wanted to call it out in case you see anything weird, but you should be good.